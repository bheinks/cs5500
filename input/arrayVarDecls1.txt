(*
	Single declarations of arrays
*)

program p;

var I : array [10..-5] of integer;
    B : array [-1..5] of boolean;
    C : array [1..50] of char;

begin
  write(5)
end.

