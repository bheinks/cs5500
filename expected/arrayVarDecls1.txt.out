

>>> Entering new scope...

+++ Adding p to symbol table with type PROGRAM

+++ Adding I to symbol table with type ARRAY 10 .. -5 OF INTEGER

+++ Adding B to symbol table with type ARRAY -1 .. 5 OF BOOLEAN

+++ Adding C to symbol table with type ARRAY 1 .. 50 OF CHAR

<<< Exiting scope...

---- Completed parsing ----
